src/ow1/heroes/baptiste.opy

```diff
@@ -29,17 +38,79 @@ rule "[baptiste.opy]: Set default immortality field cooldown":
+
+    
+
+rule "[baptiste.opy]: Start charging exo boots":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition eventPlayer.isCrouching()
+
+    eventPlayer.exo_boots_hud_visible = true
+    # don't allow the player to jump normally, because that would use the OW2 exo boots which charge faster
+    eventPlayer.setJumpEnabled(false)
+    chase(eventPlayer.exo_boots_charge_normalized, 1, duration=OW1_BAPTISTE_EXO_BOOTS_CHARGE_TIME, ChaseReeval.NONE)
+
+rule "[baptiste.opy]: Handle exo boots charge reset":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition not eventPlayer.isCrouching()
+    @Condition eventPlayer.exo_boots_charge_normalized > 0
+
+    # stop charging exo boots
+    stopChasingVariable(eventPlayer.exo_boots_charge_normalized)
+    # if the player crouches again, or jumps, or the linger time runs out, the exo boots charge goes back to 0
+    waitUntil(eventPlayer.isCrouching() or (eventPlayer.isOnGround() and eventPlayer.isHoldingButton(Button.JUMP)), OW1_BAPTISTE_EXO_BOOTS_LINGER_DURATION)
+    eventPlayer.exo_boots_hud_visible = false
+    # another grace period in case the player chose to jump, so we can use exo_boots_charge_normalized for that
+    wait()
+    eventPlayer.exo_boots_charge_normalized = 0
+    eventPlayer.setJumpEnabled(true)
+
+
+# TODO: the speed here needs to be tuned because this doesn't account for normal jump height without exo boots
+rule "[baptiste.opy]: Handle exo boots jump":
+    @Event eachPlayer
+    @Hero baptiste
+    @Condition eventPlayer.isHoldingButton(Button.JUMP)
+    @Condition eventPlayer.isOnGround()
+    @Condition eventPlayer.exo_boots_charge_normalized > 0
+    @Condition eventPlayer.isAlive()
+    @Condition not eventPlayer.hasStatusEffect(Status.ASLEEP)
+    @Condition not eventPlayer.hasStatusEffect(Status.FROZEN)
+    @Condition not eventPlayer.hasStatusEffect(Status.KNOCKED_DOWN)
+    @Condition not eventPlayer.hasStatusEffect(Status.STUNNED)
+
+    eventPlayer.applyImpulse(Vector.UP, eventPlayer.exo_boots_charge_normalized * OW1_BAPTISTE_EXO_BOOTS_MAX_JUMP_SPEED, Relativity.TO_WORLD, Impulse.CANCEL_CONTRARY_MOTION)
+
 rule "[baptiste.opy]: Use OW1 regenerative burst":
     @Event eachPlayer
     @Hero baptiste
     @Condition eventPlayer.isHoldingButton(Button.ABILITY_1)
-    @Condition eventPlayer.regen_burst_cooldown == 0
+    @Condition eventPlayer.getAbilityCooldown(Button.ABILITY_1) == 0
     @Condition eventPlayer.isAlive()
-    # TODO @Condition is not stunned slept etc
+    @Condition not eventPlayer.hasStatusEffect(Status.ASLEEP)
+    @Condition not eventPlayer.hasStatusEffect(Status.FROZEN)
+    @Condition not eventPlayer.hasStatusEffect(Status.KNOCKED_DOWN)
+    @Condition not eventPlayer.hasStatusEffect(Status.STUNNED)
 
-    eventPlayer.regen_burst_cooldown = OW1_BAPTISTE_REGEN_BURST_COOLDOWN
+    # the ability is still disabled on screen, but doing this displays the cooldown regardless which is nice
```

Comment:
Should I add some sort of help message when someone swaps to Baptiste so they are aware that the ability still works?

Reply:
I would suggest using disallowButton() instead of disableAbility().

The first essentially "cuts" the connection between the user pressing the keyboard and hero activating the ability. The second completely disables ability.

Reply:
I did something very similar for bastion's right click where I disallow the button but not the ability itself to imply that right click is still a usable ability.

Reply:
I'll definitely use that instead.